sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasQueryParameters.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Storage.Sas;
+
+namespace Azure.Storage.Files.DataLake.Sas
+{
+    /// <summary>
+    /// A <see cref="DataLakeSasQueryParameters"/> object represents the components
+    /// that make up an Azure Storage Shared Access Signature's query
+    /// parameters.  You can construct a new instance using
+    /// <see cref="DataLakeSasBuilder"/>.
+    ///
+    /// For more information, <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/constructing-a-service-sas"/>.
+    /// </summary>
+    public sealed class DataLakeSasQueryParameters : SasQueryParameters
+    {
+        /// <summary>
+        /// Gets the Azure Active Directory object ID in GUID format.
+        /// </summary>
+        public string KeyObjectId => _keyObjectId;
+
+        /// <summary>
+        /// Gets the Azure Active Directory tenant ID in GUID format
+        /// </summary>
+        public string KeyTenantId => _keyTenantId;
+
+        /// <summary>
+        /// Gets the time at which the key becomes valid.
+        /// </summary>
+        public DateTimeOffset KeyStart => _keyStart;
```

Comment:
Should be `KeyStartsOn` and `KeyExpiresOn`.

Reply:
This was shifted out from under me.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8390